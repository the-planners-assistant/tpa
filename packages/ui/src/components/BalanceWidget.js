import React, { useState } from 'react';
import styled from 'styled-components';

const StyledSlider = styled.input`
  &::-webkit-slider-runnable-track {
    ${({ theme }) => theme.slider.track}
  }
  &::-webkit-slider-thumb {
    ${({ theme }) => theme.slider.thumb}
  }
`;

const BalanceWidget = ({ considerations, onWeightChange }) => {
  const [weights, setWeights] = useState(considerations.map(c => c.weight));

  const handleChange = (index, value) => {
    const newWeights = [...weights];
    newWeights[index] = value;
    setWeights(newWeights);
    onWeightChange(considerations[index].name, value);
  };

  return (
    <div className="p-4 border rounded-lg">
      <h3 className="text-lg font-bold mb-4">Planning Balance</h3>
      <div>
        {considerations.map((consideration, index) => (
          <div key={consideration.name} className="mb-4">
            <label htmlFor={consideration.name} className="block text-sm font-medium text-gray-700">{consideration.name}</label>
            <StyledSlider
              type="range"
              id={consideration.name}
              name={consideration.name}
              min="0"
              max="100"
              value={weights[index]}
              onChange={(e) => handleChange(index, e.target.value)}
              className="w-full h-2 bg-gray-200 rounded-lg appearance-none cursor-pointer"
            />
            <div className="text-center">{weights[index]}</div>
          </div>
        ))}
      </div>
    </div>
  );
};

export default BalanceWidget;