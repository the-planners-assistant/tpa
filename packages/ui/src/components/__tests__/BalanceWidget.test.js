import React from 'react';
import { act } from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import BalanceWidget from '../BalanceWidget';
import theme from '../../../tailwind.config.js';

it('renders correctly', () => {
  const considerations = [
    { name: 'Heritage', weight: 50 },
    { name: 'Housing Delivery', weight: 70 },
  ];
  let tree;
  act(() => {
    tree = renderer.create(
      <ThemeProvider theme={theme.theme.extend}>
        <BalanceWidget considerations={considerations} onWeightChange={() => {}} />
      </ThemeProvider>
    );
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
