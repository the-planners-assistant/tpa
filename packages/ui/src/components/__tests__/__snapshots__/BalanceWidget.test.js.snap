// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders correctly 1`] = `
<div
  className="p-4 border rounded-lg"
>
  <h3
    className="text-lg font-bold mb-4"
  >
    Planning Balance
  </h3>
  <div>
    <div
      className="mb-4"
    >
      <label
        className="block text-sm font-medium text-gray-700"
        htmlFor="Heritage"
      >
        Heritage
      </label>
      <input
        className="sc-bRKDuR bbKJje w-full h-2 bg-gray-200 rounded-lg appearance-none cursor-pointer"
        id="Heritage"
        max="100"
        min="0"
        name="Heritage"
        onChange={[Function]}
        type="range"
        value={50}
      />
      <div
        className="text-center"
      >
        50
      </div>
    </div>
    <div
      className="mb-4"
    >
      <label
        className="block text-sm font-medium text-gray-700"
        htmlFor="Housing Delivery"
      >
        Housing Delivery
      </label>
      <input
        className="sc-bRKDuR bbKJje w-full h-2 bg-gray-200 rounded-lg appearance-none cursor-pointer"
        id="Housing Delivery"
        max="100"
        min="0"
        name="Housing Delivery"
        onChange={[Function]}
        type="range"
        value={70}
      />
      <div
        className="text-center"
      >
        70
      </div>
    </div>
  </div>
</div>
`;
